conf/defaults.ini

```diff
@@ -209,6 +210,41 @@ x_content_type_options = true
 # when they detect reflected cross-site scripting (XSS) attacks.
 x_xss_protection = true
 
+# Set to true to enable the COntent-security-policy header.
+# CSP allows to control resources the user agent is allowed to load
+# and help to prevent XSS atacks
+content_security_policy = true
+
+# Specifies valid sources for JavaScript
+script_src = 'unsafe-eval' 'strict-dynamic' $NONCE
+
+# Specifies valid sources for <object>, <embed>, and <applet> elements.
+object_src = "'none'"
+
+# Specifies valid sources for fonts
+font_src = "'self'"
+
+# Specifies valid sources for CSS
+style_src = 'self' 'unsafe-inline'
+
+# Specifies valid sources for Images
+img_src = '*' 'self' data:
+
+# Restricts the URLs which can be used in a document's <base> element.
+base_uri = "'self'"
+
+# Set to true to prevents loading any assets using HTTP when the page is loaded using HTTPS
+block_all_mixed_content = true
+
+#  Restricts the URLs which can be loaded using script interfaces
+connect_src = 'self' grafana.com
+
+# Specifies which manifest can be applied to the resource
+manifest_src = "'self'"
+
+# Specifies valid sources for loading media using the <audio> and <video>
+media_src = "'none'"
```

Comment:
This is just adding too much configuration to an already huge configuration file. CSP should be considered a rather advanced use case and if you're using a reverse proxy you can add all of this there. But for us to accept this I would consider having a `content_security_policy` setting of string where you can input whatever you want and by default it should not be set.

Reply:
Like this ? :
content_security_policy = script-src 'unsafe-eval' 'strict-dynamic' $NONCE; object_src = 'none'; font_src...

Reply:
It's not possible since semicolon in INI files starting a comment.
If you insist doing it that way, the only option is to change :
https://github.com/grafana/grafana/blob/master/pkg/setting/setting.go#L467 
https://github.com/grafana/grafana/blob/master/pkg/setting/setting.go#L510

from `ini.Load(filePath)` to `ini.LoadSources(LoadOptions{UnescapeValueCommentSymbols : true}, filePath)`, which allow using semicolon with escape, like this : `\;`.

Is that the an acceptable solution for you ?

Reply:
You have to wrap it in triple quotes, like [here](https://grafana.com/docs/grafana/latest/administration/configuration/#password)

Reply:
Ok, changed that to work that way :)